crates/adapters/src/catalog.rs

```diff
@@ -421,31 +426,61 @@ impl Catalog {
         }
     }
 
-    pub fn register_input_collection_handle<H>(&mut self, name: &str, handle: H)
-    where
-        H: DeCollectionHandle + 'static,
-    {
-        self.input_collection_handles
-            .insert(name.to_owned(), Box::new(handle));
+    pub fn register_input_collection_handle(&mut self, name: &str, handle: InputCollectionHandle) {
+        let name = if handle.schema.case_sensitive {
+            name.to_string()
+        } else {
+            name.to_lowercase()
+        };
+
+        self.input_collection_handles.insert(name, handle);
+    }
+
+    pub fn register_output_batch_handles(&mut self, name: &str, handles: OutputCollectionHandles) {
+        let name = if handles.schema.case_sensitive {
+            name.to_string()
+        } else {
+            name.to_lowercase()
+        };
+
+        self.output_batch_handles.insert(name, handles);
     }
 }
 
 impl CircuitCatalog for Catalog {
     /// Look up an input stream handle by name.
-    fn input_collection_handle(&self, name: &str) -> Option<&dyn DeCollectionHandle> {
+    fn input_collection_handle(&self, name: &str) -> Option<&InputCollectionHandle> {
         self.input_collection_handles
-            .get(name)
-            .map(|b| &**b as &dyn DeCollectionHandle)
+            .get(&canonical_identifier(name))
     }
 
     /// Look up output stream handles by name.
     fn output_handles(&self, name: &str) -> Option<&OutputCollectionHandles> {
-        self.output_batch_handles.get(name)
+        self.output_batch_handles.get(&canonical_identifier(name))
```

Comment:
so sometimes the metadata is in the flag "case_sensitive", and sometimes it's in the name itself (quotes). Not easy to keep track which is which. Could we have different types for them, or is that overkill?

Reply:
The only time it comes from the flag is when retrieving it from the schema, maybe I should just add a convenience read function for that, which happens to be the same function you and @lalithsuresh  are asking for in your other comments.